manim/utils/tex_file_writing.py

```diff
@@ -211,22 +210,19 @@ def convert_to_svg(dvi_file, extension, page=1):
 
     Returns
     -------
-    :class:`str`
+    :class:`Path`
         Path to generated SVG file.
     """
-    result = dvi_file.replace(extension, ".svg")
-    result = Path(result)
-    result_str = result.as_posix()
-    dvi_file = Path(dvi_file).as_posix()
+    result = dvi_file.with_suffix(".svg")
     if not result.exists():
         commands = [
             "dvisvgm",
-            "--pdf" if extension == ".pdf" else "",
+            "--pdf" if dvi_file.suffix == ".pdf" else "",
```

Comment:
I think this change should be made separately (along with refactoring the variable name from `dvi_file` to something that captures that the file can be an xdv or pdf as well).

```suggestion
            "--pdf" if extension == ".pdf" else "",
```